### DISCUSSION TITLE
How do I debug rust and ts at the same time？

### DISCUSSION THREAD

**Discussion by moushicheng** [rolldown/rolldown#674] [Contribution]
i use javascript debug termial, but when i enter ``rolldown`` or ``build.write``, it cant not enter rust logic

gif⬇️
![20240327151039_rec_](https://github.com/rolldown/rolldown/assets/54257412/1d62b93f-ed7e-41e0-a50a-814ecd723a48)

**Comment by hyfdev** [comment]
Probably will write a guidance in the future but not now. You could get some hints on https://web-infra-dev.github.io/rspack-dev-guide/debugging/mix-debug.html.

**Reply by moushicheng** [reply]
thanks! if i find Mixed Debugging Way, let me support guidance work❤️

**Reply by moushicheng** [reply]
@hyf0 hey! i found debug way in rolldown,thx rspack-dev-guide you provide for me.
Details:
https://github.com/rolldown/rolldown/pull/678

**Comment by moushicheng** [comment]
## Debug with Javascript and Rust
1. Open Javascript Debug Termial
2. Execute any program that works with javascript and rust in Javascript Debug Termial
3. Start Attach:Rust in Tab Run and Debug
4. select javascript process.
5. enjoy mixed debug!
I think you'll still be confused after you read this steps.so you can see this video to learn about this process

[bilibili.com: Debug with Javascript and Rust](https://www.bilibili.com/video/BV1Rm421n79f/)
